### PR TITLE
add custom_api directory with create and update

### PR DESCRIPTION
**Pull request by GideonBature**
completed the task by adding the `custom_api` directory and setting up the `POST` and `PUT` method for it. Also, added the `api_url` and `regex` fields to the `QuestTaskDocument` in the `models.rs` file. Also made changes to all the files using the `QuestTaskDocument` to include the `api_url` and the `regex` fields, setting their values as None, since they are not needed in those other context, at least in this scope. I also did this to remove likely errors that might emanate from those files due to the changes I made to the `QuestTaskDocument` model in the `models.rs` file.

### CODE REVIEW

File: src/endpoints/admin/custom_api/create_custom_api.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        let last_id = doc.id;
+        next_id = last_id + 1;
+    }
+
+    // Build a vector of FieldElement from the comma separated contracts string
+    let parsed_contracts: Vec<FieldElement> = body
+        .contracts
+        .split(",")
+        .map(|x| FieldElement::from_str(&x).unwrap())
+        .collect();
```
Comment (Marchand-Nicolas) [inline src/endpoints/admin/custom_api/create_custom_api.rs]: This part is unused

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
Missing `pub mod custom_api` in `src\endpoints\admin\mod.rs`.

File: src/endpoints/mod.rs [hunk 2]
```diff
@@ -17,3 +17,4 @@ pub mod leaderboard;
 pub mod quest_boost;
 pub mod quests;
 pub mod unique_page_visit;
+pub mod custom_api;
```
Comment (Marchand-Nicolas) [inline src/endpoints/mod.rs]: This is not the right mod. There i a mod.rs in each folder. This is to indicate rust all the dependency tree. Every time, when adding a folder or a file, we need to update the mod.rs that is in the same directory. In your case, you added your folder in `src\endpoints\admin`, so you have to update `src\endpoints\admin\mod.rs`

File: src/endpoints/admin/custom_api/mod.rs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+pub mod create_custom_api;
+pub mod update_custom_api;
```
Comment (Marchand-Nicolas) [inline src/endpoints/admin/custom_api/mod.rs:2]: Here this mod.rs is setup perfectly

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
Very little change!

**Comment by Marchand-Nicolas** [comment]
Also, I suggest you to try installing cargo so that you can try compiling or running the project, because all these issues are compiling issues, not even runtime issues

**Comment by GideonBature** [comment]
Noted, sorry, an oversight from my own end, Thank you ser.

**Review by Marchand-Nicolas — APPROVED** [review]
Working really well! Lgtm

**Pull request merged**